plugins/pulp_rpm/plugins/db/models.py

```diff
@@ -480,6 +487,120 @@ def rpm_search_dicts(self):
+            return all(x == y for x, y in zip(existing_packages, new_packages))
+        return False
+
+    def merge_errata(self, other):
+        """
+        Merge two errata with the same errata_id.
+
+        There are two parts:
+        - merging of the pkglists in case of the erratum with the same id in different repositories
+        - overwriting the erratum metadata based on the `updated` field
+
+        NOTE: The second part should be eliminated after we change the way erratum is stored in the
+        MongoDB.
+
+        :param other: The erratum we are combining with the existing one
+        :type  other: pulp_rpm.plugins.db.models.Errata
+
+        :return: the erratum after merge
+        :rtype: pulp_rpm.plugins.db.models.Errata
+        """
+        self.merge_pkglists(other)
+        if self.update_needed(other):
+            for field_name in self.erratum_fields_update:
+                setattr(self, field_name, getattr(other, field_name))
+        return self
+
+    def update_needed(self, other):
+        """
+        Decide based on the `updated` field if the update of the existing erratum is needed.
+
+        :param other: potentially a newer version of the erratum
+        :type  other: pulp_rpm.plugins.db.models.Errata
+
+        :return: True if the other erratum is newer than the existing one
+        :rtype:  bool
+        """
+        # The `updated` field is just a string in the MongoDB, so there is no strict format for this
+        # date-time field. If we are not able to parse the `updated` field either in existing
+        # erratum or in the new erratum, the metadata of existing erratum won't be updated.
+
+        err_msg = 'Fail to update the %s erratum %s: Unable to parse the `updated` field'
+        existing_err_msg = err_msg % ('existing', self.errata_id)
+        other_err_msg = err_msg % ('uploaded', self.errata_id)
+        try:
+            existing_updated_dt = util.errata_format_to_datetime(self.updated, msg=existing_err_msg)
+            new_updated_dt = util.errata_format_to_datetime(other.updated, msg=other_err_msg)
+        except ValueError as e:
+            for msg in e.args:
+                _LOGGER.warn(msg)
+        else:
+            return new_updated_dt > existing_updated_dt
+        return False
+
+    def merge_pkglists(self, other):
+        """
+        Merge pkglists of the two errata.
+
+        :param other: The erratum we are combining with the existing one
+        :type  other: pulp_rpm.plugins.db.models.Errata
+
+        """
+        # Merging of pkglists:
+        #   - add _pulp_repo_id to old collection if packages are the same
+        #   - update existing collection if the other collection is newer and from the same
+        #     repository
+        #   - otherwise add a new collection
+
+        # map (collection_name, _pulp_repo_id) to collection index in the pkglist of existng erratum
+        existing_pkglist_idx_map = dict(
```

Comment:
```
existing_pkglist_map = {}
for p in self.pkglist:
    package_name = p['name']
    package_repo_id = p['_pulp_repo_id']
    existing_pkglist_map[package_name] = package_repo_id
```

It's more lines but much easier to read. I also renamed the variable to `existing_pkglist_map`.